website/auth.py

```diff
@@ -36,21 +39,23 @@ def logout():
     return redirect(url_for('auth.login'))
 
 
-@auth.route('/sign-up', methods=['GET', 'POST'])
-def sign_up():
+@auth.route('/permissions', methods=['GET', 'POST'])
+def permissions():
+    
 
     if request.method == 'POST':
+        
         email = request.form.get('email')
-        first_name = request.form.get('first-name')
-        last_name = request.form.get('last-name')
+        first_name = request.form.get('first_name')
+        last_name = request.form.get('last_name')
         password = request.form.get('password')
-        confirm_password = request.form.get('confirm-password')
-        role = request.form.get('role')
-        print(role)
+        role = request.form.get('roles')
 
         emailList = email.split('@')
-
-        user = User.query.filter_by(email=email).first()
+        try:
+            user = User.query.filter_by(email=email).first()
+        except:
+            user = None
         if user:
```

Comment:
This is where I ran into one issue. Make sure when you're testing that you aren't testing the signup with the same email you have already used, or it will not allow you to add it to the database. When I used a made-up email, `123@colby.edu`, it worked fine!